### PR TITLE
Fixed buggy behavior when trying to edit a generated column #fixed

### PR DESCRIPTION
**Pull request by gduh**
## fixed:
- Fix the issue #1155 
It was'nt so easy to precisely reproduce the issue and find what was wrong, but finally the fix was easy.

2 things : 
- I added an alert view when the user tries to edit a generated column. The alert occurs only  when the user modifies a first generated field.
- This issue already existed in at least one case (Solved in this PR too) : 
   - Preferences --> Alerts & Logs --> Check "Show warning before executing a query"
   - Start to edit a field value
   - Click on another table name in the left panel
   - An alert view ask you if you want to edit the row. Say Cancel
   - same issue...

Closes: #1159 

## Tested:
- Processors:
  - [X] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [X] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
- Localizations:
  - [X] English
  - [ ] Spanish
  - [ ] Other (please specify)
- Xcode Version:12.5.1 (12E507)

### CODE REVIEW

File: Source/Views/TableViews/SPCopyTable.m [hunk 1]
```diff
@@ -1314,6 +1314,25 @@ - (BOOL)shouldUseFieldEditorForRow:(NSUInteger)rowIndex column:(NSUInteger)colIn
 	return NO;
 }
 
+/**
+ * Determine whether to use the sheet for editing; do so if the multipleLineEditingButton is enabled,
+ * or if the column was a blob or a text, or if it contains linebreaks.
+ */
```
Comment (Jason-Morcos) [inline Source/Views/TableViews/SPCopyTable.m:1320]: Comment here looks to not be associated with the function

File: Source/Views/TableViews/SPCopyTable.m [hunk 2]
```diff
@@ -1314,6 +1314,25 @@ - (BOOL)shouldUseFieldEditorForRow:(NSUInteger)rowIndex column:(NSUInteger)colIn
...
+- (BOOL)shouldNotEditGeneratedFieldForColumn:(NSUInteger)colIndex
```
Comment (Jason-Morcos) [inline Source/Views/TableViews/SPCopyTable.m:1321]: Maybe we can tweak the name and flip it to positive - ex. isColunmnEditable maybe?

**Review by Jason-Morcos — COMMENTED** [review]
Love it! Just a few little things

File: Source/Views/TableViews/SPCopyTable.m (hunk 1 above)
Comment (gduh) [reply Source/Views/TableViews/SPCopyTable.m]: Oops! Reviewed comment....

File: Source/Views/TableViews/SPCopyTable.m (hunk 2 above)
Comment (gduh) [reply Source/Views/TableViews/SPCopyTable.m]: Done....
I left the current alert message as it is the only control for now (Only check generated column). But in the future, if there are other checks for the editable column, the message should be reconsidered.

**Review by Jason-Morcos — APPROVED** [review]
Looks great!

**Comment by Jason-Morcos** [comment]
Pushed out a new beta with this change, @gduh!

One thought - should we say the name of the column instead of `the current field` in `The current field is a generated column, you cannot edit it!`?
Maybe something like, `"astrological_sign" is a generated column and therefore cannot be edited.`?

**Comment by gduh** [comment]
Yes, I think this message is better too. 
I'm currently on vacation, I'll change it on my return.

**Pull request merged**